demos/motion_tracking/main_motion_tracking.py

```diff
@@ -0,0 +1,142 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+from threading import Thread
+from functools import partial
+from bokeh.plotting import figure, curdoc
+from bokeh.layouts import gridplot, Spacer
+from bokeh.models import LinearColorMapper, ColorBar, Title, Button
+from bokeh.models.ranges import DataRange1d
+import sys
+from multiprocessing import Pipe
+from motion_tracking_network import MotionNetwork
+
+# ==========================================================================
+# Parameters
+# ==========================================================================
+recv_pipe, send_pipe = Pipe()
+num_steps = 10
+
+# Checks whether terminate button has been clicked and allows to stop
+# updating the bokeh doc
+is_done = [False]
```

Comment:
Why is this a list of bools?

Reply:
It needs to be something mutable such that it changes in place when we later set it to True in the stop_callback. Like this it can also be accessed by the main loop. Alternatively we could also declare it as a global variable.

Reply:
But why a list? Is it not always one value? Could it not just be a bool?
I'm probably missing something here.

Reply:
It you have it as a value it would not be mutuable. Its scope in the stop_calback would therefore be local, so that the main_loop wouldn't know that the demo has finished.